code/edgeidentity/src/test/java/com/adobe/marketing/mobile/edge/identity/IdentityStateTests.java

```diff
@@ -21,223 +20,230 @@
+			.thenReturn(
+				new SharedStateResult(
+					SharedStateStatus.SET,
+					Collections.singletonMap(
+						IdentityConstants.SharedState.IdentityDirect.ECID,
+						fetchedIdentityDirectECID.toString() // not the same persisted Identity ECID for the sake of tests
+					)
+				)
+			);
+
+		assertTrue(identityState.bootupIfReady(mockSharedStateCallback));
+		assertNotNull(identityState.getIdentityProperties().getECID());
+		assertEquals(persistedDirectIdentityECID, identityState.getIdentityProperties().getECID());
+		verify(mockIdentityStorageManager).savePropertiesToPersistence(identityState.getIdentityProperties());
+		verify(mockSharedStateCallback)
+			.createXDMSharedState(identityState.getIdentityProperties().toXDMData(false), null);
+	}
+
+	@Test
+	public void testBootUpIfReady_waitsForIdentityDirectStateIfRegistered_stateIsNone() {
+		final IdentityState identityState = new IdentityState(mockIdentityStorageManager);
+		final Map<String, Object> hubSharedState = new HashMap<>();
+		hubSharedState.put(
+			"extensions",
+			new HashMap<String, Object>() {
+				{
+					put(
+						"com.adobe.module.identity",
+						new HashMap<String, String>() {
+							{
+								put("friendlyName", "Identity");
+								put("version", "2.0.0");
+							}
+						}
+					);
+				}
+			}
+		);
 
-		// test
-		state.bootupIfReady(mockSharedStateCallback);
-		verify(mockSharedPreferenceEditor, never()).apply();
+		when(mockSharedStateCallback.getSharedState(IdentityConstants.SharedState.Hub.NAME, null))
+			.thenReturn(new SharedStateResult(SharedStateStatus.SET, hubSharedState));
+		when(mockSharedStateCallback.getSharedState(IdentityConstants.SharedState.IdentityDirect.NAME, null))
+			.thenReturn(new SharedStateResult(SharedStateStatus.NONE, null));
 
-		// verify
-		assertEquals(persistedProps.getECID(), state.getIdentityProperties().getECID());
-		assertEquals(1, setXDMSharedEventStateCalledTimes);
+		assertFalse(identityState.bootupIfReady(mockSharedStateCallback));
+		assertNull(identityState.getIdentityProperties().getECID());
+		verify(mockIdentityStorageManager, never()).savePropertiesToPersistence(identityState.getIdentityProperties());
+		verify(mockSharedStateCallback, times(0)).createXDMSharedState(any(), any());
 	}
 
 	@Test
-	public void testBootupIfReady_whenIdentityDirectRegistered_onFirstBoot_waitsForIdentityDirectECID() {
-		// setup
-		Mockito
-			.when(mockContext.getSharedPreferences(IdentityConstants.DataStoreKey.IDENTITY_DIRECT_DATASTORE_NAME, 0))
-			.thenReturn(mockSharedPreference);
-		Mockito
-			.when(mockSharedPreference.getString(IdentityConstants.DataStoreKey.IDENTITY_DIRECT_ECID_KEY, null))
-			.thenReturn(null);
-
-		IdentityState state = new IdentityState(new IdentityProperties());
-
-		// test
-		hubSharedState = new HashMap<>();
+	public void testBootUpIfReady_waitsForIdentityDirectStateIfRegistered_stateIsPending() {
```

Comment:
nit: `stateStatusIsPending`

Reply:
reword: `testBootUpIfReady_waitsForIdentityDirectStateIfRegistered` pending is confusing with the actual sharedStateStatus value.